candump.c

```diff
@@ -615,23 +635,15 @@ int main(int argc, char **argv)
 
 	while (running) {
 
-		FD_ZERO(&rdfs);
-		for (i=0; i<currmax; i++)
-			FD_SET(s[i], &rdfs);
-
-		if (timeout_current)
-			*timeout_current = timeout_config;
-
-		if ((ret = select(s[currmax-1]+1, &rdfs, NULL, NULL, timeout_current)) <= 0) {
-			//perror("select");
+		if ((num_events = epoll_wait(fd_epoll, events_pending, currmax, timeout_ms)) <= 0) {
+			//perror("epoll_wait");
 			running = 0;
 			continue;
 		}
 
-		for (i=0; i<currmax; i++) {  /* check all CAN RAW sockets */
-
-			if (FD_ISSET(s[i], &rdfs)) {
-
+		for (i=0; i<num_events; i++) {  /* check waiting CAN RAW sockets */
+			struct if_info* obj = (struct if_info*)events_pending[i].data.ptr;
+			{ /* this was an if(FD_ISSET), bracket kept just to keep the diff small, no indentation change */
```

Comment:
please remove the extra level of indention, you can use `git diff --ignore-space-change` to simplify the diff while reviewing.

Reply:
Done, makes it nicer.